pkg/composer/serviceparser/build.go

```diff
@@ -82,6 +83,29 @@ func parseBuildConfig(c *types.BuildConfig, project *types.Project, imageName st
 		}
 	}
 
+	for _, s := range c.Secrets {
+		fileRef := types.FileReferenceConfig(s)
+		if err := identifiers.Validate(fileRef.Source); err != nil {
+			return nil, fmt.Errorf("secret source %q is invalid: %w", fileRef.Source, err)
+		}
+		projectSecret, ok := project.Secrets[fileRef.Source]
+		if !ok {
+			return nil, fmt.Errorf("build: secret %s is undefined", fileRef.Source)
+		}
+		var src string
+		if filepath.IsAbs(projectSecret.File) {
+			logrus.Warnf("build.secrets should be relative path, got %q", projectSecret.File)
+			src = projectSecret.File
+		} else {
+			src = filepath.Join(ctxDir, projectSecret.File)
```

Comment:
```suggestion
			src, err = securejoin.SecureJoin(ctxDir, projectSecret.File)
```

to block  ../../../… attacks